classlib/src/main/java/org/teavm/classlib/java/util/CollectionsFactory.java

```diff
@@ -0,0 +1,92 @@
+/*
+ *  Copyright 2014 Alexey Andreev.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.teavm.classlib.java.util;
+
+import static org.teavm.classlib.java.util.TObjects.requireNonNull;
+import org.teavm.classlib.java.lang.TIllegalArgumentException;
+import org.teavm.classlib.java.lang.TNullPointerException;
+
+/**
+ * Factory-methods for List/Set/Map.of(...).
+ */
+class CollectionsFactory {
+
+    private CollectionsFactory() {
+    }
+
+    /**
+     * Create an immutable list for the {@code List.of(...)} factory methods.
+     *
+     * @throws TNullPointerException if any element is null
+     */
+    @SafeVarargs
+    static <E> TList<E> createList(E... elements) {
+        if (elements == null || elements.length == 0) {
+            return TCollections.emptyList();
+        }
+
+        final TList<E> list = new TArrayList<>();
+        for (E element : elements) {
+            list.add(requireNonNull(element, "element"));
+        }
+
+        return TCollections.unmodifiableList(list);
+    }
+
+    /**
+     * Create an immutable set for the {@code Set.of(...)} factory methods.
+     *
+     * @throws TNullPointerException     if any element is null
+     * @throws TIllegalArgumentException if duplicate elements are given
+     */
+    @SafeVarargs
+    static <E> TSet<E> createSet(E... elements) {
+        if (elements == null || elements.length == 0) {
+            return TCollections.emptySet();
+        }
+
+        final TSet<E> set = new THashSet<>();
+        for (E element : elements) {
+            if (!set.add(requireNonNull(element, "element"))) {
```

Comment:
Is such behaviour documented for Set.of? What about tests which test this behaviour? What about tests at all?

Reply:
Yes, all the factory methods reject null values, Set.of doesn't permit duplicate values and Map.of/ofEntries doesn't permit duplicate keys. (The classes are also supposed to be serializable, but I guess, this doesn't apply to TeaVM.)

https://docs.oracle.com/javase/9/docs/api/java/util/List.html#immutable
https://docs.oracle.com/javase/9/docs/api/java/util/Set.html#immutable
https://docs.oracle.com/javase/9/docs/api/java/util/Map.html#immutable

I can add tests to verify this behaviour.